### PR TITLE
fix: Template v1 and Conversion CRD stanzas v0.37.x

### PR DESCRIPTION
**Pull request by rschalo**
Fixes #N/A 

**Description**
Webhook E2Es failed because they were expecting to be able to find references to v1 during smoke test cleanup and could not.

**How was this change tested?**
Make e2etest

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/hack/e2e_scripts/install_karpenter.sh [hunk 1]
```diff
@@ -1,7 +1,12 @@
 aws eks update-kubeconfig --name "$CLUSTER_NAME"
 
-# Parse minor version to determine whether to enable the webhooks
-K8S_VERSION_MINOR="${K8S_VERSION#*.}"
+# First, conditionally install the webhook stanza and CRDs
+if (( "$WEBHOOKS_ENABLED" == 'false' )); then
+helm upgrade --install karpenter-crd oci://public.ecr.aws/karpenter/karpenter-crd \
+  --namespace kube-system \
+  --version "0-$(git rev-parse HEAD)" \
+  --set webhook.enabled=${WEBHOOKS_ENABLED}
```
Comment (engedaam) [inline test/hack/e2e_scripts/install_karpenter.sh:8]: Should we add `--wait`?

Comment (rschalo) [reply test/hack/e2e_scripts/install_karpenter.sh:8]: What do we get with it? I also need to change the version, that's not correct as written.

File: test/hack/e2e_scripts/install_karpenter.sh [hunk 2]
```diff
@@ -1,7 +1,13 @@
 aws eks update-kubeconfig --name "$CLUSTER_NAME"
 
-# Parse minor version to determine whether to enable the webhooks
-K8S_VERSION_MINOR="${K8S_VERSION#*.}"
+# First, conditionally install the webhook stanza and CRDs
+if (( "$WEBHOOKS_ENABLED" == 'false' )); then
+helm upgrade --install karpenter-crd oci://public.ecr.aws/karpenter/karpenter-crd \
```
Comment (jonathan-innis) [inline test/hack/e2e_scripts/install_karpenter.sh:5]: Can we just use the same upgrade command and either always pass `webhook.enabled` or construct the arguments and append this one on if it's false?

File: test/hack/e2e_scripts/install_karpenter.sh [hunk 3]
```diff
@@ -1,7 +1,13 @@
...
+  --namespace kube-system \
+  --version "0.37.5" \
```
Comment (jonathan-innis) [inline test/hack/e2e_scripts/install_karpenter.sh:7]: This is pinning to a version that may change in the future with future patch releases (unless you were just doing this for testing)

File: hack/mutation/ec2nodeclasses.sh [hunk 4]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+
+VERSION_START="$(cat charts/karpenter-crd/templates/karpenter.k8s.aws_ec2nodeclasses.yaml | yq '.spec.versions.[0] | line')"
+VERSION_END="$(cat charts/karpenter-crd/templates/karpenter.k8s.aws_ec2nodeclasses.yaml | yq '.spec.versions.[1] | line')"
+VERSION_END=$(($VERSION_END+1))
+
+cat charts/karpenter-crd/templates/karpenter.k8s.aws_ec2nodeclasses.yaml | awk -v n=$VERSION_START 'NR==n {sub(/$/,"\n{{- if .Values.webhook.enabled }}")} 1' \
+| awk -v n=$VERSION_END 'NR==n {sub(/$/,"\n{{- end }}")} 1' > ec2nc.yaml
+
+cat ec2nc.yaml > charts/karpenter-crd/templates/karpenter.k8s.aws_ec2nodeclasses.yaml && rm ec2nc.yaml
```
Comment (jonathan-innis) [inline hack/mutation/ec2nodeclasses.sh:10]:
```suggestion
TEMP=$(mktemp)
cat charts/karpenter-crd/templates/karpenter.k8s.aws_ec2nodeclasses.yaml | awk -v n=$VERSION_START 'NR==n {sub(/$/,"\n{{- if .Values.webhook.enabled }}")} 1' \
| awk -v n=$VERSION_END 'NR==n {sub(/$/,"\n{{- end }}")} 1' > $TEMP

cat $TEMP > charts/karpenter-crd/templates/karpenter.k8s.aws_ec2nodeclasses.yaml
```

Comment (jonathan-innis) [reply hack/mutation/ec2nodeclasses.sh:10]: This prevents you from having to do the delete

**Review by jonathan-innis — APPROVED** [review]
LGTM 🚀

**Pull request merged**